src/Integration.UnitTests/Notifications/SonarQubeNotificationServiceTests.cs

```diff
@@ -238,41 +238,62 @@ public void StartAsync_Throws_ArgumentNullException_For_ProjectKey()
                     date: newerEventDate,
                     message: "foo");
 
+            var event3 = new SonarQubeNotification(
+                    category: "QUALITY_GATE",
+                    link: new Uri("http://foo.com"),
+                    date: newestEventDate,
+                    message: "foo");
+            
             serverNotifications.Add(event1);
+
+            // 1. Start monitoring
+            // Must call StartAsync so the cancellation token is initialized. This will fetch the event data.
+            await notifications.StartAsync("anykey", null);
+
+            sonarQubeServiceMock.Verify(x => x.GetNotificationEventsAsync("anykey", It.IsAny<DateTimeOffset>(), It.IsAny<CancellationToken>()), Times.Once);
+            notifications.GetNotificationData().LastNotificationDate.Should().Be(olderEventDate);
+
+            // 2. Add newer events and simulate the timer firing
             serverNotifications.Add(event2);
+            serverNotifications.Add(event3);
 
-            // Act
+            // Simulate the timer triggering
             timerMock.Raise(x => x.Elapsed += null, (ElapsedEventArgs)null);
 
-            // Wait some time because the timer event is not awaitable
-            await Task.Delay(100);
```

Comment:
I don't think it is necessary to _await_ here: the mocks are using _Task.FromResult_ to return completed tasks so the code is running synchronously.